src/libraries/Microsoft.Extensions.Options/gen/Emitter.cs

```diff
@@ -405,5 +445,19 @@ private StaticFieldInfo GetOrAddStaticValidator(ref Dictionary<string, StaticFie
 
             return staticValidatorInstance;
         }
+
+        /// <summary>
+        /// Returns a non-randomized hash code for the given string.
+        /// We always return a positive value.
+        /// </summary>
+        internal static int GetNonRandomizedHashCode(string s)
+        {
+            uint result = 2166136261u;
+            foreach (char c in s)
+            {
+                result = (c ^ result) * 16777619;
+            }
+            return Math.Abs((int)result);
+        }
```

Comment:
We might want to consider using SHA256 here instead of FNV-1a. I _think_ that's what the compiler itself uses when it needs uniqueness.

Reply:
After offline discussion, we decided to stick with `FNV-1a`.